### PR TITLE
modify spectralnorm

### PR DESCRIPTION
**Pull request by wangna11BD**
### PR types
Function optimization

### PR changes
APIs

### Describe
modify spectral_norm to improve the usability of spectral_norm

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+import collections
+import paddle
+import paddle.nn as nn
+import paddle.fluid as fluid
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:22]: 下面单测用2.0API吧

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import paddle.fluid.core as core
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:23]: core下面没用到，去掉这行

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            v_norm = np.sqrt((v * v).sum())
+            v = v / (v_norm + eps)
+            u = np.matmul(weight_mat, v)
+            u_norm = np.sqrt((u * u).sum())
+            u = u / (u_norm + eps)
+        sigma = (u * np.matmul(weight_mat, v)).sum()
+        return weight / sigma
+
+    def test_check_output(self):
+        # fluid.enable_imperative()
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:71]: remove un-used code

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        sn = spectral_norm(
+            linear,
+            n_power_iterations=self.n_power_iterations,
+            eps=self.eps,
+            dim=self.dim)
+        u = sn.weight_u.numpy().copy()
+        v = sn.weight_v.numpy().copy()
+        outputs = []
+        for name, data in self.data.items():
+            output = linear(fluid.dygraph.to_variable(data))
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:92]: paddle.to_tensor

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        self.data_desc = (['x', [2, 3, 3]], )
+        self.n_power_iterations = 1
+        self.eps = 1e-12
+        self.dim = None
+
+
+class TestDygraphWeightNormWithIterations(TestDygraphSpectralNorm):
+    def init_test_case(self):
+        self.batch_size = 1
+        self.data_desc = (['x', [2, 3, 3]], )
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:117]: shape调大些

File: python/paddle/nn/utils/spectral_norm_hook.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserve.
```
Comment (qingqing01) [inline python/paddle/nn/utils/spectral_norm_hook.py:1]: 2021

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:1]: 2021

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py (hunk 5 above)
Comment (wangna11BD) [reply python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:117]: 大了单侧会超时，然后失败

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py (hunk 1 above)
Comment (wangna11BD) [reply python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:22]: done

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py (hunk 2 above)
Comment (wangna11BD) [reply python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:23]: done

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py (hunk 3 above)
Comment (wangna11BD) [reply python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:71]: done

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py (hunk 4 above)
Comment (wangna11BD) [reply python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:92]: done

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py (hunk 5 above)
Comment (wangna11BD) [reply python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:117]: done

File: python/paddle/nn/utils/spectral_norm_hook.py (hunk 6 above)
Comment (wangna11BD) [reply python/paddle/nn/utils/spectral_norm_hook.py:1]: done

File: python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py (hunk 7 above)
Comment (wangna11BD) [reply python/paddle/fluid/tests/unittests/test_dygraph_spectral_norm.py:1]: done

**Pull request closed without merging**